src/reducers/builder.reducer.js

```diff
@@ -104,6 +104,17 @@ const reducer = (state = initialState, action) => {
         ...state,
         locations: action.locations
       };
+
+    case 'ADD_LOCATION_SUCCESS': {
+      const newLocations = state.locations;
```

Comment:
In the variable newLocations, you wrote down the reference to the array. You need to use the spread operator.